Sources/AWSLambdaRuntime/Lambda+Codable.swift

```diff
@@ -78,6 +78,64 @@ internal struct CodableVoidClosureWrapper<In: Decodable>: LambdaHandler {
     }
 }
 
+// MARK: - Async
+
+#if compiler(>=5.5) && $AsyncAwait
```

Comment:
this is good stuff. should we also mark the closure ones deprecated? I would argue we should.